### PR TITLE
DS-3905 Implement a generic uuid lookup endpoint (DSpaceObjectService based)

### PR DESCRIPTION
**Pull request by abollini**
This is a revision of #2176 to use a DSpaceObjectService to lookup for the UUID instead than SOLR merging the approach of #2176 with the original one in #2125

### CODE REVIEW

**Review by tomdesair — COMMENTED** [review]
Looks good and the functionality is covered by integrations tests.

I only have one small (non-blocking) remark.

**Review by tdonohue — APPROVED** [review]
👍 This looks great to me.

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/UUIDLookupRestController.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            throws IOException, SQLException, SearchServiceException {
+
+        Context context = null;
+        try {
+            context = ContextUtil.obtainContext(request);
+            for (DSpaceObjectService dSpaceObjectService : contentServiceFactory.getDSpaceObjectServices()) {
+                DSpaceObject dso = dSpaceObjectService.find(context, uuid);
+                if (dso != null) {
+                    DSpaceObjectRest dsor = converter.convert(dso);
+                    URI link = linkTo(dsor.getController(), dsor.getCategory(), English.plural(dsor.getType()))
```
Comment (tomdesair) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/UUIDLookupRestController.java]: Can you use `getTypePlural()` here? https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/RestModel.java#L35

Comment (abollini) [reply dspace-spring-rest/src/main/java/org/dspace/app/rest/UUIDLookupRestController.java]: done

**Review by tomdesair — APPROVED** [review]
Thx!

**Pull request merged**